### PR TITLE
Header Component updates - part 1

### PR DESCRIPTION
**Pull request by GHaberis**
## What does this change?

- Update headers `__config.js` to include full set of links
- Renames subcomponents of  `Header`, replace references to `SubNav` and `SecondarySubNav` with `MainMenu`
- Add accessibility attributes to the `Header` component and all subcomponents, tested with keyboard navigation
- Add no-js (checkbox) fallback to the `OpenMainMenu` component
- Add types to all `Header` component and all subcomponents

What next?
- Test with screen reader.
- Replace hardcoded `px` with `grid`.
- Remove hardcoded color codes
- Add Jobs/Dating/Holidays/Masterclasses links
- Add edition picker
- Enable Subscribe/Find a job/Sign in/Search navigation

### CODE REVIEW

**Comment by GHaberis** [comment]
**New Component Tree**
```
Header/index.js
Header/Nav/index.js
Header/Nav/Logo.js
Header/Nav/__config__.js
Header/Nav/Pillars/index.js
Header/Nav/Pillars/Pillar.js
Header/Nav/OpenMainMenu/index.js
Header/Nav/OpenMainMenu/VeggieBurger.js
Header/Nav/MainMenu/index.js
Header/Nav/MainMenu/MainMenuColumns/index.js
Header/Nav/MainMenu/MainMenuColumns/index.js
Header/Nav/MainMenu/MainMenuColumns/MainMenuColumnButton.js
Header/Nav/MainMenu/MainMenuColumns/MainMenuColumnLinks/index.js
Header/Nav/MainMenu/MainMenuColumns/MainMenuColumnLinks/MainMenuColumnLink/index.js
Header/Nav/MainMenu/MainMenuColumns/MainMenuColumnLinks/MainMenuLinkTitle.js
```
**Old Component Tree**
```
Header/index.js
Header/Nav/index.js
Header/Nav/Logo.js
Header/Nav/SubNavLink.js
Header/Nav/VeggieBurger.js
Header/Nav/Pillars/index.js
Header/Nav/Pillars/Pillar.js
Header/Nav/Pillars/__config__.js
Header/Nav/SubNav/index.js
Header/Nav/SubNav/SubNavList/index.js
Header/Nav/SubNav/SubNavList/SubNavListItem/index.js
Header/Nav/SubNav/SubNavList/SubNavListItem/SubNavButton.js
Header/Nav/SubNav/SubNavList/SubNavListItem/SecondarySubNavList/index.js
Header/Nav/SubNav/SubNavList/SubNavListItem/SecondarySubNavList/SecondarySubNavListItem/index.js
Header/Nav/SubNav/SubNavList/SubNavListItem/SecondarySubNavList/SecondarySubNavListItem/SubNavTitle.js
```

File: sites/frontend/components/Header/Nav/MainMenu/MainMenuColumns/MainMenuColumn/MainMenuColumnLinks/MainMenuLink/MainMenuLinkTitle.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+// @flow
+import { styled } from '@guardian/guui';
+
+import { tablet, desktop } from '@guardian/pasteup/breakpoints';
+import { egyptian } from '@guardian/pasteup/fonts';
+import { pillars } from '@guardian/pasteup/palette';
+
+import type { ColumnType } from '../../../../../../Nav/__config__';
```
Comment (SiAdcock) [inline sites/frontend/components/Header/Nav/MainMenu/MainMenuColumns/MainMenuColumn/MainMenuColumnLinks/MainMenuLink/MainMenuLinkTitle.js]: Maybe we should try to avoid relative paths, which can be hard to follow. Do we not resolve paths from, say, `components`?

File: sites/frontend/components/Header/Nav/OpenMainMenu/index.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+import { desktop } from '@guardian/pasteup/breakpoints';
+import { screenReaderOnly } from '@guardian/pasteup/mixins';
+import { headline } from '@guardian/pasteup/fonts';
+
+import VeggieBurger from './VeggieBurger';
+
+const ScreenReadable = styled('span')(screenReaderOnly);
+
+const navPrimaryColour = '#121212';
+const navSecondaryColour = '#5d5f5f';
```
Comment (SiAdcock) [inline sites/frontend/components/Header/Nav/OpenMainMenu/index.js]: Maybe these can move to `pasteup`? We'll probably use them in the slim nav, and maybe other places

File: sites/frontend/components/Header/Nav/OpenMainMenu/index.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
+// @flow
+import { styled, Component } from '@guardian/guui';
+
+import { desktop } from '@guardian/pasteup/breakpoints';
+import { screenReaderOnly } from '@guardian/pasteup/mixins';
+import { headline } from '@guardian/pasteup/fonts';
+
+import VeggieBurger from './VeggieBurger';
+
+const ScreenReadable = styled('span')(screenReaderOnly);
```
Comment (SiAdcock) [inline sites/frontend/components/Header/Nav/OpenMainMenu/index.js]: Should we name this component `ScreenReaderOnly`?

File: sites/frontend/components/Header/Nav/OpenMainMenu/index.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+                aria-controls={ariaControls}
+                tabindex="-1"
+                key="OpenMainMenuCheckbox"
+                role="menuitemcheckbox"
+            />,
+        ];
+    }
+}
+
+export default OpenMainMenu;
```
Comment (SiAdcock) [inline sites/frontend/components/Header/Nav/OpenMainMenu/index.js]: `OpenMainMenu` sounds like a verb rather than a noun. What does this component represent? The Main Menu button / call to action?

File: sites/frontend/components/Header/Nav/MainMenu/MainMenuColumns/MainMenuColumn/MainMenuColumnLinks/MainMenuLink/MainMenuLinkTitle.js (hunk 1 above)
Comment (GHaberis) [reply sites/frontend/components/Header/Nav/MainMenu/MainMenuColumns/MainMenuColumn/MainMenuColumnLinks/MainMenuLink/MainMenuLinkTitle.js]: We don't right now, but I agree relative paths are a nightmare to keep maintained when moving folder around. I can update the `webpack` config to make all import paths relative to `components`. I guess maybe we didn't do this to begin with as the vast majority of imports are sibling components or child components, but in this case  it's a 'config' we're having to go back up the tree to retrieve.

File: sites/frontend/components/Header/Nav/OpenMainMenu/index.js (hunk 2 above)
Comment (GHaberis) [reply sites/frontend/components/Header/Nav/OpenMainMenu/index.js]: Agreed, but I'd like to raise a separate PR where I tidy up the colour codes

File: sites/frontend/components/Header/Nav/OpenMainMenu/index.js (hunk 4 above)
Comment (GHaberis) [reply sites/frontend/components/Header/Nav/OpenMainMenu/index.js]: Yep, `OpenMainMenu` could render as the Main Menu Button OR the Main Menu Checkbox if JS is not running, that's why I didn't call it `MainMenuButton`. What do you think of main `MainMenuToggle`,  "toggle" can be a noun right?

Comment (SiAdcock) [reply sites/frontend/components/Header/Nav/OpenMainMenu/index.js]: Ha, more English language ambiguity! Yeah, I think `Toggle` is reasonable

File: sites/frontend/components/Header/Nav/MainMenu/MainMenuColumns/MainMenuColumn/MainMenuColumnLinks/MainMenuLink/MainMenuLinkTitle.js (hunk 1 above)
Comment (SiAdcock) [reply sites/frontend/components/Header/Nav/MainMenu/MainMenuColumns/MainMenuColumn/MainMenuColumnLinks/MainMenuLink/MainMenuLinkTitle.js]: Right on. Let's fix it when this PR is merged
